src/rules/newline-after-import.js

```diff
@@ -140,7 +140,7 @@ module.exports = {
             line: node.loc.end.line,
             column,
           },
-          message: `Expected ${options.count} empty line${options.count > 1 ? 's' : ''} after import statement not followed by another import.`,
+          message: `Expected ${options.count} empty line${options.count > 1 ? 's' : ''} after ${type} statement not followed by another ${type}.`,
```

Comment:
can we make `type` (and while we're at it, the count and the pluralization) part of the `data` instead of interpolated in?

it'd also be fine to do this as a followup.

Reply:
I'm not sure I understand exactly. Is it correct to say that create a function that params are count ,type and return the created message and apply it to other messages? below is example code

```
function example(count, type){
    return  `Expected ${options.count} empty line${options.count > 1 ? 's' : ''} after ${type} statement not followed by another ${type}.`,
}
....
messages:example(options.count,type)

```

Reply:
@ljharb If my understanding is correct, I will proceed in the follow-up PR.

Reply:
No, not a function - using eslint's own message interpolation format; no function required. A followup is fine.

Reply:
Ah, I understand. Are you referring to the code format below? (example is `consistent-type-specifier-style` rule)

```javascript
context.report({
....
            message: 'Prefer using a top-level {{kind}}-only import instead of inline {{kind}} specifiers.',
            data: {
              kind: kind.join('/'),
            },
          }
 ....
```

Reply:
yes, exactly that.

Reply:
Thank you. I will proceed with the follow-up work as mentioned above.